tests/lib/state.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+remove_users_from_state(){
+    systemctl stop snapd.service
+    cp /var/lib/snapd/state.json ./state.json.back
```

Comment:
Can we please separate this into an explicit action called `save_state` used on prepare, name the file as `state.json.saved`, and have that as an independent pair of `restore_state`? Then it's more explicit and clear when the state is saved and when it's restored.  The removal function can then do just that: remove users from state.